R/install_sys_deps.R

```diff
@@ -0,0 +1,186 @@
+addToProfile = function(line, type = 'OS') {
+  paths = if (type == 'OS') c('.zshrc', '.bashrc', '.profile') else '.Rprofile'
+  paths = paste0('~/', paths)
+
+  for (path in paths) {
+    lines = if (file.exists(path)) readLines(path) else character()
+    if (!(line %in% lines)) write(line, path, append = TRUE)}
+
+  invisible()}
+
+
+installSraToolkit = function(installDir, addToPath = TRUE) {
+  sraPath = path.expand(file.path(installDir, 'sratoolkit', 'bin'))
+
+  # Check if sratoolkit already exists at location
+  if (dir.exists(sraPath)) {
+    cat('SRA Toolkit already installed, skipping...\n')
+
+  } else {
+    cat('Installing SRA Toolkit...\n')
+    # Change working directory
+    withr::local_dir(installDir)
+
+    # Determine latest available SRA Toolkit version
+    url = 'https://ftp-trace.ncbi.nlm.nih.gov/sra/sdk/'
+    raw = RCurl::getURL(url)
+    x = strsplit(raw, '\\n')[[1L]]
+    m = regexpr('[0-9]\\.[0-9]\\.[0-9]', x)
+    sraVersion = max(regmatches(x, m))
+
+    # Based on operating system, download correct version to correct path
+    sraOsTar = if (Sys.info()[['sysname']] == 'Darwin') 'mac64' else 'ubuntu64'
+    sraTar = glue('sratoolkit.{sraVersion}-{sraOsTar}.tar.gz')
+
+    # Download based on version and OS, unzip, then rename directory
+    withr::local_file(sraTar)
+    download.file(glue('{url}{sraVersion}/{sraTar}'), sraTar, quiet = TRUE)
+    utils::untar(sraTar)
+    . = file.rename(sub('\\.tar\\.gz', '', sraTar), 'sratoolkit')}
```

Comment:
The `file.xxx()` functions return a value non-invisibly, so that's my convention of indicating I don't care about the returned value, I just don't want it printed to the screen.

Reply:
Any reason why not to wrap it in `invisible()`?

Reply:
Fewer characters per line.